OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/operations/impl/OperationRepo.kt

```diff
@@ -345,6 +335,26 @@ internal class OperationRepo(
         }
     }
 
+    /**
+     * Stall processing the queue so the backend's DB has to time
+     * reflect the change before we do any other operations to it.
+     * NOTE: Future: We could run this logic in a
+     * coroutineScope.launch() block so other operations not
+     * effecting this these id's can still be done in parallel,
+     * however other parts of the system don't currently account
+     * for this so this is not safe to do.
+     */
+    suspend fun delayForPostCreate(hasIDTranslation: Boolean) {
+        val postCreateDelay: Long = _configModelStore.model.opRepoPostCreateDelay
+        if (!hasIDTranslation || postCreateDelay == 0L) {
```

Comment:
Remove check for `postCreateDelay == 0L to simply this code`.
* It's valid to `delay(0)`, it behaves just a like `yield` which is ok in this scenarion.